### PR TITLE
Change configurator to DataClass and polish manager after changes

### PR DESCRIPTION
**Pull request by Sllambias**
Much of this is also just a cleanup - removing files from OmniFrame. Some might be added later but I will get them from the OmniFrame repo as needed. I've also cleaned up old naming conventions which will require some adaptation with e.g. imports, but I believe it's gonna be super nice long term!

### CODE REVIEW

**Comment by Sllambias** [comment]
I am rewriting tests atm to support our new dataclasses

**Comment by Sllambias** [comment]
Tested so far: 

- [ ] task_convert
- [ ] preprocessing
- [ ] fresh training
- [ ] resumed training
- [ ] training with alternative manager (YuccaManager_AdamW)
- [ ] inference
- [ ] 2D and 3D

to be tested:
- [ ] finetuning
- [ ] alternative architectures
- [ ] others?

File: yucca/training/configuration/configure_paths_and_plans.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+from dataclasses import dataclass
+from typing import Union
+from yucca.paths import yucca_models, yucca_preprocessed_data
+from yucca.preprocessing.UnsupervisedPreprocessor import UnsupervisedPreprocessor
+from yucca.preprocessing.ClassificationPreprocessor import ClassificationPreprocessor
+from yucca.utils.files_and_folders import recursive_find_python_class
+
+
+@dataclass
+class PathAndPlanConfig:
```
Comment (asbjrnmunk) [inline yucca/training/configuration/configure_paths_and_plans.py:13]: This dataclass has become very big. Wouldn't it maybe make sense to split into PathConfig and PlanConfig?

File: yucca/training/lightning_modules/YuccaLightningModule.py [hunk 2]
```diff
@@ -26,7 +25,7 @@ def __init__(
         self,
         config: dict,
         learning_rate: float = 1e-3,
-        loss_fn: str = "DiceCE",
+        loss_fn: str = None,
```
Comment (asbjrnmunk) [inline yucca/training/lightning_modules/YuccaLightningModule.py:28]: Why should this be none?

**Review by asbjrnmunk — COMMENTED** [review]
Cool! Overall I think it looks good, however I think we should consider splitting the config into two. What do you think?

Also I would have been very grateful if the Config and the manager polish would have been in two different PR's, since 45 changed files with different changes can be a little much, when this is not absolutely needed.

File: yucca/training/lightning_modules/YuccaLightningModule.py (hunk 2 above)
Comment (Sllambias) [reply yucca/training/lightning_modules/YuccaLightningModule.py:28]: Because then we can set the default loss based on the task type (if you scroll down a little bit) so for task types segmentation and classification we use DiceCE as default but for unsupervised it defaults to MSE. 

Ofc, if you set the loss_fn when you call the LightningModule it will just use whatever you specify

File: yucca/training/configuration/configure_paths_and_plans.py (hunk 1 above)
Comment (Sllambias) [reply yucca/training/configuration/configure_paths_and_plans.py:13]: Yea I was also thinking about this when I was finishing it up. I'll do that!

File: tests/test_configurations.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    split_config = get_split_config(train_data_dir=path_and_plan_config.train_data_dir, task=path_and_plan_config.task)
+    assert split_config is not None
+
+    callback_config = get_callback_config(
+        task=path_and_plan_config.task,
+        save_dir=path_and_plan_config.save_dir,
+        version_dir=path_and_plan_config.version_dir,
+        version=path_and_plan_config.version,
+    )
+    assert callback_config is not None
```
Comment (asbjrnmunk) [inline tests/test_configurations.py:31]: We should be able to make some more asserts here on the contents of the configs. There are many use cases and ideally each should have its own test case. At a minimum I would suggest:

1. Split the test case up into one for each of path+plan, input dims, split, callbacks. 
2. Test the content of the config (eg. does the path_and_plan_config contain the right save_dir?)

**Comment by Sllambias** [comment]
Yea I'm sorry about the big PR. I know it's heavy, but it just felt weird to rewrite tests and change imports knowing I would have to re-do everything 2 minutes later in a new PR for polishing but I probably should've just pushed the config and left the tests and imports

File: tests/test_configurations.py (hunk 3 above)
Comment (jakobamb) [reply tests/test_configurations.py:31]: good call

**Comment by Sllambias** [comment]
image
Dataclasses good

**Review by jakobamb — COMMENTED** [review]
really nice, see minor comments

File: yucca/training/lightning_modules/__init__.py
[the remark this answers is not in the document]
Comment (Sllambias) [reply yucca/training/lightning_modules/__init__.py:1]: the init file?

File: yucca/training/augmentation/YuccaAugmentationComposer.py [hunk 4]
```diff
@@ -30,10 +30,11 @@ def __init__(
         patch_size: list | tuple,
         is_2D: bool = False,
         parameter_dict: dict = {},
+        use_preset_for_task_type: str = None,
```
[the remark this answers is not in the document]
Comment (Sllambias) [reply yucca/training/augmentation/YuccaAugmentationComposer.py:33]: Good idea

File: yucca/training/configuration/configure_plans.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+from typing import Union
+from yucca.paths import yucca_models, yucca_preprocessed_data
+from yucca.preprocessing.UnsupervisedPreprocessor import UnsupervisedPreprocessor
+from yucca.preprocessing.ClassificationPreprocessor import ClassificationPreprocessor
+from yucca.utils.files_and_folders import recursive_find_python_class
+
+
+@dataclass
+class PlanConfig:
+    ckpt_path: str
```
Comment (jakobamb) [inline yucca/training/configuration/configure_plans.py:14]: I think ckpt_path can be None?

**Review by jakobamb — CHANGES_REQUESTED** [review]

File: yucca/training/lightning_modules/__init__.py
[the remark this answers is not in the document]
Comment (jakobamb) [reply yucca/training/lightning_modules/__init__.py:1]: sorry, I was referring to setting:

os.environ["OMP_NUM_THREADS"] = "1" in this init

File: yucca/training/configuration/configure_plans.py (hunk 5 above)
Comment (Sllambias) [reply yucca/training/configuration/configure_plans.py:14]: True

File: yucca/training/configuration/configure_plans.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+from typing import Union
+from yucca.paths import yucca_models, yucca_preprocessed_data
+from yucca.preprocessing.UnsupervisedPreprocessor import UnsupervisedPreprocessor
+from yucca.preprocessing.ClassificationPreprocessor import ClassificationPreprocessor
+from yucca.utils.files_and_folders import recursive_find_python_class
+
+
+@dataclass
+class PlanConfig:
+    ckpt_path: Union[str, None]
```
Comment (asbjrnmunk) [inline yucca/training/configuration/configure_plans.py:14]: I would suggest moving this to the PathConfig.

**Review by asbjrnmunk — APPROVED** [review]
LTGM!
